server/graphql/mutations/orders.js

```diff
@@ -368,4 +368,23 @@ export function updateSubscription(remoteUser, args) {
     return updatePromise
         .then(() => order.update({ PaymentMethodId: newPm.id}))
   })
-}
\ No newline at end of file
+}
+
+export async function refundTransaction(_, args, req) {
+  // 0. Retrieve transaction from database
+  const transaction = await models.Transaction.findById(args.id, {
+    include: [models.Order, models.PaymentMethod]
+  });
+
+  // 1. Verify user permission. User must be either
+  //   a. User that created transaction (within 24h) -- Not implemented yet
+  //   b. Host Collective receiving the donation -- Not implemented yet
+  //   c. Site Admin
+  if (!req.remoteUser.isRoot()) {
+    throw new errors.Unauthorized({ message: "Not a site admin" });
+  }
+
+  // 2. Refund via payment method
+  // 3. Create new transactions with the refund value in our database
+  return libPayments.refundTransaction(transaction);
```

Comment:
What will happen if transaction wasn't found?

Reply:
I was not handling that case. Just did it in the latest commits! Thanks for the catch!